### PR TITLE
Create Radio button

### LINKED ISSUE #31: Radio button component

### PR DESCRIPTION
**Pull request by bramvanhoutte**
Closes #31

### CODE REVIEW

File: src/components/Radio/RadioItem/RadioItem.module.css [hunk 1]
```diff
@@ -0,0 +1,102 @@
+.radioInput {
+  height: 0;
+  width: 0;
+  opacity: 0;
+  position: absolute;
+}
+
+.radioInput + .radioButtonContainer {
```
Comment (RobinWijnant) [inline src/components/Radio/RadioItem/RadioItem.module.css:8]: Why did you add `.radioInput +` in front?

File: src/components/Radio/RadioItem/RadioItem.module.css [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  align-items: center;
+  justify-content: center;
+  margin-right: 14px;
+}
+
+.radioInput + .radioButtonContainer .header .checkBox .checkBoxFilling {
+  width: 0px;
+  height: 0px;
+  border-radius: 50%;
+  background-color: var(--font-color-sub);
```
Comment (RobinWijnant) [inline src/components/Radio/RadioItem/RadioItem.module.css:43]: I think this was by mistake. A font color is used as background color here.

File: src/components/Radio/RadioItem/RadioItem.module.css [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+.radioInput + .radioButtonContainer .header .title {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: flex-start;
+}
+
+.radioInput:disabled + .radioButtonContainer .header .title {
```
Comment (RobinWijnant) [inline src/components/Radio/RadioItem/RadioItem.module.css:53]: Nice, the way you did this 👏

File: src/components/Radio/RadioItem/RadioItem.module.css [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+.radioInput + .radioButtonContainer .description {
+  color: var(--font-color-sub);
+  font-size: 13px;
+  margin: 0px 0px 0px 30px;
+}
+
+.radioButtonContainer:hover {
+  background-color: var(--color-primary);
```
Comment (RobinWijnant) [inline src/components/Radio/RadioItem/RadioItem.module.css:64]: If the primary color is blue, the font color should also change to white or something.
Maybe add `color: var(--color-font-primary-font)`

To prevent this in the future we could create a theme selector in the docs to preview the component for another theme. Should I add an issue for this?

**Comment by RobinWijnant** [comment]
I think I have found an issue. When only using the keyboard, you can use space to select the radio that is in focus but when you hit tab on the keyboard to go to the next radio, the selected radio gets unselected

File: src/index.css [hunk 5]
```diff
@@ -1,5 +1,13 @@
 @import 'theme.css';
 
+html {
+  box-sizing: border-box;
```
[the remark this answers is not in the document]
Comment (bramvanhoutte) [reply src/index.css:4]: Yes, I get that but the opposite is also true. If they already configured this (which is pretty much the standard), then our component will look different then it is supposed to. 

This is not what they had in mind. Their enthusiasm for Ventura now goes from 9 to 4/10.

File: src/components/Radio/RadioItem/RadioItem.module.css (hunk 1 above)
Comment (bramvanhoutte) [reply src/components/Radio/RadioItem/RadioItem.module.css:8]: Because a lot of the selectors contain .radioInput:disabled or radioInput:checked. To me this made it more readable to keep the same basic structure.

File: src/components/Radio/RadioItem/RadioItem.module.css (hunk 4 above)
Comment (bramvanhoutte) [reply src/components/Radio/RadioItem/RadioItem.module.css:64]: Yes! Good idea!

File: src/components/Radio/RadioItem/RadioItem.module.css (hunk 2 above)
Comment (bramvanhoutte) [reply src/components/Radio/RadioItem/RadioItem.module.css:43]: Good catch

File: src/components/Radio/RadioGroup/RadioGroup.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  name,
+  groupRef,
+}: Props) => {
+  const [selectedValue, setSelectedValue] = useState(defaultValue);
+
+  const onRadioItemChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSelectedValue(e.target.value);
+    if (onChange) {
+      onChange(e);
+    }
```
[the remark this answers is not in the document]
Comment (bramvanhoutte) [reply src/components/Radio/RadioGroup/RadioGroup.tsx:33]: I know.
But I think using a simple if-statement is more readable. What is your opinion?

File: src/components/Radio/Radio.mdx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import { Radio, Loader } from '../../index.ts'
+
+# Radio
+
+## Examples
+
+### Basic usage
+
+<Playground>
+  <Radio.Group name="classes1" defaultValue="react">
```
[the remark this answers is not in the document]
Comment (bramvanhoutte) [reply src/components/Radio/Radio.mdx:17]: Yes! I do this because otherwise multiple RadioGroups will have the same name value.

This doesn't cause errors. It just won't give the wanted result. However, this should maybe be specified in the documentation.

File: src/components/Radio/Radio.mdx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    <Radio.Item value="react" label="Introduction To React" />
+    <br />
+    <Radio.Item value="docker" label="Introduction To Docker" />
+    <br />
+    <Radio.Item value="python" label="Introduction To Python" />
+  </Radio.Group>
+</Playground>
+
+### Radio button label
```
[the remark this answers is not in the document]
Comment (bramvanhoutte) [reply src/components/Radio/Radio.mdx:26]: Yes, I should make the documentation a little clearer.

File: src/components/Radio/RadioItem/RadioItem.module.css (hunk 4 above)
Comment (bramvanhoutte) [reply src/components/Radio/RadioItem/RadioItem.module.css:64]: This is a hard one however. Because when not selected the background is white. Meaning white text will not work

**Comment by RobinWijnant** [comment]
> think I have found an issue. When only using the keyboard, you can use space to select the radio that is in focus but when you hit tab on the keyboard to go to the next radio, the selected radio gets unselected

I tested this while there were two radio groups with the same name. This caused the issue so this is my bad

File: src/index.css (hunk 5 above)
[the remark this answers is not in the document]
Comment (RobinWijnant) [reply src/index.css:4]: We will create an issue to apply styles to all components, isolated from the code base where it is implemented. See #38

File: src/components/Radio/RadioItem/RadioItem.module.css (hunk 4 above)
Comment (RobinWijnant) [reply src/components/Radio/RadioItem/RadioItem.module.css:65]: See #36
